trestle/transforms/implementations/tanium.py

```diff
@@ -494,21 +487,76 @@ def analysis(self) -> List[str]:
         analysis.append(f'observations: {len(self.observations)}')
         return analysis
 
+    def _get_local_definitions(self, system_component: SystemComponent) -> LocalDefinitions1:
+        """Get local definitions."""
+        rval = LocalDefinitions1()
+        for component in self.components:
+            if component.uuid == system_component.uuid:
+                rval.components = [component]
+                rval.inventory_items = []
+                for inventory_item in self.inventory:
+                    for implemented_component in inventory_item.implemented_components:
+                        if implemented_component.component_uuid == system_component.uuid:
+                            rval.inventory_items.append(inventory_item)
+                break
+        return rval
+
+    def _get_observation_subject_uuids(self, observation: Observation) -> List[str]:
+        """Get subject uuids for given observation."""
+        rval = []
+        for subject in observation.subjects:
+            rval.append(subject.subject_uuid)
+        return rval
+
+    def _get_local_definitions_uuids(self, local_definitions: LocalDefinitions1) -> List[str]:
+        """Get inventory uuids for given local definitions."""
+        rval = []
+        for inventory_item in local_definitions.inventory_items:
+            rval.append(inventory_item.uuid)
+        return rval
+
+    def _is_matching_uuid(self, observation_subject_uuids: List[str], local_definitions_uuids: List[str]) -> bool:
+        """Check if any given observation uuid is present in given local definition uuids."""
+        rval = False
+        for subj_uuid in observation_subject_uuids:
+            if subj_uuid in local_definitions_uuids:
```

Comment:
can use set intersection:
`return set(a).intersection(set(b)) != set()`
or something like that
where a and b are lists

Reply:
Fixed.